### PR TITLE
[fix] Improve engines line alignment and formatting

### PR DESCRIPTION
**Pull request by allixx**
## What does this PR do?

Fixes vertical alignment of engines and cached/proxied links. Adds missing space between cached and proxied links.

Replaces seemingly useless left padding of engines div with top padding. Having top padding here really helps to add a bit of visual separation of engines line (it often gets quite crowded with multiple engines) and search result.

## Why is this change important?

Those pixel discrepancies really hurt the eye. No visual separation between cached and proxied link is a cherry on top.

Before:
before

After:
after

### CODE REVIEW

**Review by Bnyro — APPROVED** [review]
LGTM!

**Comment by Bnyro** [comment]
Due to the CSS changes, you would also need to run `make static.build.commit`.

**Comment by return42** [comment]
@allixx with you last [force push](https://github.com/searxng/searxng/compare/5f0e29fa04962f8934887c38c77e844951d592a9..d3583942cbb1723ab01a25942dcd97ef03468272) you also changed (silently) the padding in the `.engine` class ... was this change intended?

image

**Comment by allixx** [comment]
Yep, I've documented it in updated pull request description.
`Replaces seemingly useless left padding of engines div with top padding. Having top padding here really helps to add a bit of visual separation of engines line (it often gets quite crowded with multiple engines) and search result.`

**Comment by return42** [comment]
> Yep, I've documented it in updated pull request description.

Thanks for clarify .. for the future; please add a note in the thread .. otherwise reviewers won't notify that new changes have been added .. 

BTW this means any previous approval needs to be renewed (@Bnyro).

File: searx/templates/simple/macros.html [hunk 1]
```diff
@@ -42,7 +42,7 @@ <h3>{{ result_link(result.url, result.title|safe) }}</h3>
 {%- macro result_sub_footer(result, proxify) -%}
 <div class="engines">
   {% for engine in result.engines %}<span>{{ engine }}</span>{% endfor %}
-  {{ result_link(cache_url + result.url, icon_small('ellipsis-vertical-outline') + _('cached'), "cache_link") }}&lrm; {% if proxify and proxify_results %} {{ result_link(proxify(result.url), icon('link') + _('proxied'), "proxyfied_link") }} {% endif %}
+  {{ icon_small('ellipsis-vertical-outline') + result_link(cache_url + result.url, _('cached'), "cache_link") }}&lrm; {% if proxify and proxify_results %}&emsp;{{ result_link(proxify(result.url), icon('link') + _('proxied'), "proxyfied_link") }} {% endif %}
```
Comment (return42) [inline searx/templates/simple/macros.html]: The `&emsp;` is responsible for a (IMO) to great space between cache and proxy link

image

I haven't look at the details, but may we get a better result, when we use a `<span>` tag around the links (like we use one line above to enumerate the engines)?

Comment (allixx) [reply searx/templates/simple/macros.html]: Whoops, my bad. It was meant to be `&ensp;` thanks for catching this. Will fix.

Regarding wrapping the whole line in a container, I didn't have much success with this approach. Icon within `<a>` breaks vertical alignment with or without container for me, so I moved it outside.

**Review by return42 — APPROVED** [review]
Thanks! :+1: 

-----

FYI: config to test was ..

```diff
diff --git a/searx/settings.yml b/searx/settings.yml
index ebd6d5463..0ceee4488 100644
--- a/searx/settings.yml
+++ b/searx/settings.yml
@@ -114,7 +114,7 @@ ui:
   # center the results ?
   center_alignment: false
   # URL prefix of the internet archive, don't forget trailing slash (if needed).
-  # cache_url: "https://webcache.googleusercontent.com/search?q=cache:"
+  cache_url: "https://webcache.googleusercontent.com/search?q=cache:"
   # Default interface locale - leave blank to detect from browser information or
   # use codes from the 'locales' config section
   default_locale: ""
@@ -144,12 +144,12 @@ ui:
 # morty proxy the key is base64 encoded (keep the !!binary notation)
 # Note: since commit af77ec3, morty accepts a base64 encoded key.
 #
-# result_proxy:
-#   url: http://127.0.0.1:3000/
+result_proxy:
+  url: http://127.0.0.1:3000/
 #   # the key is a base64 encoded string, the YAML !!binary prefix is optional
 #   key: !!binary "your_morty_proxy_key"
 #   # [true|false] enable the "proxy" button next to each result
-#   proxify_results: true
+  proxify_results: true
 
 # communication with search engines
 #
```

**Comment by Bnyro** [comment]
I've been testing this again, and I think some additional spacing between the three dots and the "cached" link would be nice here. For reference, how it currently looks at my browser:

2024-01-31T15:43:19,041859660+01:00

An equal spacing from "qwant" to the three dots, and from the three dots to "cached" would looks slightly more natural imo.

**Comment by allixx** [comment]
Personally I don't have a preference, if you guys think this will look better, I'll make it so.

**Comment by allixx** [comment]
Looks clean enough, I guess.
after

**Review by Bnyro — APPROVED** [review]

**Review by return42 — APPROVED** [review]
> Looks clean enough, I guess.

yeah looks much better / thanks a lot for taking the time :+1:

**Pull request merged**